src/pretix/control/views/waitinglist.py

```diff
@@ -366,3 +427,87 @@ def get_success_url(self) -> str:
             'event': self.request.event.slug,
             'organizer': self.request.event.organizer.slug
         })
+
+class EntryUpdate(EventPermissionRequiredMixin, UpdateView):
+    model = WaitingListEntry
+    template_name = 'pretixcontrol/waitinglist/update.html'
+    permission = 'can_change_orders'
+    form_class = WaitingListEntryEditForm
+    context_object_name = 'entry'
+
+
+
+    #def dispatch(self, request, *args, **kwargs):
+        #self.fields['subevent'] = WaitingListEntry.objects.filter(event=self.request.event)
+    #    return super().dispatch(request, *args, **kwargs)
+    def get_queryset(self):
+        return WaitingListEntry.objects.filter(
+            event=self.request.event,
+        )
+    def get_object(self, queryset=None) -> WaitingListEntry:
+        try:
+            return self.request.event.waitinglistentries.get(
+                id=self.kwargs['entry'],
+                event=self.request.event,
+                voucher__isnull=True,
+            )
+        except WaitingListEntry.DoesNotExist:
+            raise Http404(_("The requested entry does not exist."))
+
+    def get_success_url(self) -> str:
+        return reverse('control:event.orders.waitinglist', kwargs={
+            'event': self.request.event.slug,
+            'organizer': self.request.event.organizer.slug
+        })
+
+    @transaction.atomic
+    def update(self, request, *args, **kwargs):
```

Comment:
Where does this `update` method come from? I do not think this works/is called at all. I might be wrong, but shouldn’t that be `form_valid`? I haven’t looked to deep into this yet.

Reply:
Look at e.g. `DiscountUpdate` https://github.com/pretix/pretix/blob/master/src/pretix/control/views/discounts.py#L96-L136
There is also a `form_invalid` method.